src/Parser.php

```diff
@@ -218,12 +224,28 @@ protected function parseFeature()
         while ($this->predictTokenType() !== 'EOS') {
             $node = $this->parseExpression();
 
-            if (is_string($node)) {
+            if (is_string($node) && $this->compatibilityMode->shouldRemoveFeatureDescriptionPadding()) {
                 $text = preg_replace('/^\s{0,' . ($token['indent'] + 2) . '}|\s*$/', '', $node);
                 $description .= ($description !== null ? "\n" : '') . $text;
                 continue;
             }
 
+            if (is_string($node)) {
+                if ($node === "\n" && $description === null) {
+                    // Ignore empty lines before the start of the description
+                    continue;
+                }
+
+                // It must be part of the feature description (text & newlines later in the document will be consumed as
+                // part of parsing Background / Scenario before execution returns to this loop).
+                $description .= $node;
+                if ($node !== "\n") {
+                    // Text nodes do not end with a newline, add one. The final trailing newline is rtrimmed below.
+                    $description .= "\n";
+                }
```

Comment:
This is a little convoluted because for a feature file like:

```gherkin
Feature: Extra table content
  Tables are delimited by pipes on both sides.
  Anything that isn't enclosed is not part of
  the table.

  It is not recommended to use this feature, but
  it is how the implementation currently works.
```

The Lexer gives us the following tokens (note that the empty line is represented by "\n", slightly inconsistently to the lines with text).

```
"  Tables are delimited by pipes on both sides."
"  Anything that isn't enclosed is not part of"
"  the table."
"\n"
"  It is not recommended to use this feature, but"
"  it is how the implementation currently works."
"\n"
```